app/src/androidTest/java/org/mozilla/fenix/ui/NavigationToolbarTest.kt

```diff
@@ -138,4 +141,46 @@ class NavigationToolbarTest {
             verifyFindNextInPageResult("1/1")
         }.closeFindInPage { }
     }
+
+    @Test
+    fun verifySecurePageSecuritySubMenuTest() {
+        val defaultWebPage = "https://mozilla-mobile.github.io/testapp/loginForm"
+        val defaultWebPageTitle = "Login_form"
+        val connectionSecurity = "Connection is secure"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(defaultWebPage.toUri()) {
+        }.openSiteSecuritySheet {
+            verifyQuickActionSheet(defaultWebPage.substring(8, 32), connectionSecurity)
+            openSecureConnectionSubMenu(connectionSecurity)
+            verifySecureConnectionSubMenu(defaultWebPageTitle, defaultWebPage, connectionSecurity)
+        }
+    }
+
+    @Test
+    fun verifyInsecurePageSecuritySubMenuTest() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        val connectionSecurity = "Connection is not secure"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(defaultWebPage.url) {
+        }.openSiteSecuritySheet {
+            verifyQuickActionSheet(defaultWebPage.url.toString().substring(7, 16), connectionSecurity)
```

Comment:
See above

Reply:
Switched to tryGetHostFromUrl() and seems to be working fine.